### DISCUSSION TITLE
Console: List Databases

### DISCUSSION THREAD

**Discussion by gramian** [ArcadeData/arcadedb#716] [Ideas]
Dear Developers,

the console is (IMHO) missing a command to list databases. The console mostly provides commands that have a corresponding HTTP API endpoint plus extra console commands. However, while there is a `/api/v1/databases` endpoint listing the databases at the current database path, a similar command (i.e. `list`) for the console is not available. I think this would be a useful addition to the console capabilities.

**Comment by lvca** [comment]
I agree, this command is missing in the console and is very easy to implement, namely:
1. list all the directory under the configured database directory (global setting for it)
2. check if in the directory there is a schema.json. If it's there, that should be a database folder, so the name of the folder is the database name

**Reply by gramian** [reply]
I agree for the most part. Yet for the remote connection, the user's "allowed databases" need to be determined, as for the endpoint, right?

**Reply by lvca** [reply]
Correct. If the remote connection has been opened, then it should be delegated to the server with the existing GET `/databases`. If local, then a directory list is ok
